### PR TITLE
Ci speedup

### PR DESCRIPTION
**Pull request by byronz**
this PR allows we control the percy snapshot finalize process by pytest flag, so we can do `pytest --percyfinalize` in case we need parallelism in the integration jobs.

### CODE REVIEW

File: dash/testing/plugin.py [hunk 1]
```diff
@@ -30,7 +30,15 @@ def pytest_addoption(parser):
...
-        "--headless", action="store_true", help="Run tests in headless mode"
+        "--headless",
+        action="store_true",
+        help="set this flag to run in headless mode",
+    )
+
+    dash.addoption(
+        "--percyfinalize",
+        action="store_false",
+        help="set this flag to control percy finalize at CI level",
```
Comment (alexcjohnson) [inline dash/testing/plugin.py]: would this make more sense to be called `--percynofinalize`? `"store_false"` just seems inherently confusing...

Comment (byronz) [reply dash/testing/plugin.py]: `--nopercyfinalize`?

Comment (alexcjohnson) [reply dash/testing/plugin.py]: > `--nopercyfinalize`

even better!

**Comment by byronz** [comment]
dancer without actual approve can't get merged lol

**Pull request merged**